lib/view_component/compiler.rb

```diff
@@ -163,6 +163,18 @@ def template_errors
               "There can only be a template file or inline render method per variant."
           end
 
+          uniq_variants = variants.compact.uniq
```

Comment:
I happened to be poking around Bundler the other day and saw these rules:

```
- Any - characters in a host name are mapped to a triple dash (___) in the corresponding environment variable.
- Any . characters in a host name are mapped to a double dash (__) in the corresponding environment variable.
```

(Above https://bundler.io/man/bundle-config.1.html#CONFIGURE-BUNDLER-DIRECTORIES).

What do you think about using this naming approach to avoid collisions?

Reply:
@joelhawksley I think that's a great idea! I'll update the pull request.

Reply:
@joelhawksley I've updated the pull request so method names replace dashes with double underscores. Dots aren't replaced with triple underscores , though, because currently using dots wouldn't work because the code getting variants and handlers from filenames assumes there are only two dots in the filename. I intend to try to add support for dots in a few weeks, although I can't confirm for sure that I'll be able to do so.

In the meantime, more comments regarding this pull request are welcome :wink:.